api/tests/Unit/Controller/SelfRegisterControllerTest.php

```diff
@@ -266,6 +265,169 @@ public function throwErrorForValidCaseNumberButDetailsNotMatching()
                     'created_at' => $now->format('c'),
                 ],
             ],
+            'matching_errors' => [
+                'client_lastname' => true,
+                'deputy_lastname' => true,
+                'deputy_postcode' => true,
+            ],
+        ];
+
+        $this->assertEquals($expectedErrorJson, json_decode($responseArray['message'], true));
+    }
+
+    /**
+     * @test
+     */
+    public function throwErrorForValidCaseNumberClientLastnameDeputyPostcodeButInvalidDeputyLastname()
+    {
+        $now = new DateTime();
+
+        $casRec = new PreRegistration([
+            'Case' => '97643164',
+            'ClientSurname' => 'Douglas',
+            'DeputyUid' => 'DEP00199',
+            'DeputySurname' => 'Murphy',
+            'DeputyAddress1' => 'Victoria Road',
+            'DeputyPostcode' => 'SW1',
+            'ReportType' => 'OPG102',
+            'MadeDate' => '2010-03-30',
+            'OrderType' => 'pfa',
+            'NDR' => 'yes',
+        ], $now);
+
+        $this->fixtures()->persist($casRec);
+        $this->fixtures()->flush();
```

Comment:
There's some repetition in this, the previous and next test in regards to setting up data that could be refactored. Giving it a descriptive function name will also make the intent clear on what the data is.

Reply:
That's a good shout. Will look into refactoring this.